WordPress/src/main/java/org/wordpress/android/ui/posts/RemotePreviewLogicHelper.kt

```diff
@@ -1,24 +1,25 @@
 package org.wordpress.android.ui.posts
 
 import android.app.Activity
+import dagger.Reusable
 import org.wordpress.android.fluxc.model.PostModel
 import org.wordpress.android.fluxc.model.SiteModel
 import org.wordpress.android.ui.ActivityLauncherWrapper
 import org.wordpress.android.ui.WPWebViewUsageCategory
-import org.wordpress.android.ui.uploads.UploadUtils
-import org.wordpress.android.ui.uploads.UploadUtils.PostUploadAction
-import org.wordpress.android.ui.uploads.UploadUtils.PostUploadAction.REMOTE_AUTO_SAVE
-import org.wordpress.android.ui.uploads.UploadUtils.PostUploadAction.UPLOAD
-import org.wordpress.android.ui.uploads.UploadUtils.PostUploadAction.UPLOAD_AS_DRAFT
+import org.wordpress.android.ui.uploads.UploadActionUseCase
+import org.wordpress.android.ui.uploads.UploadActionUseCase.UploadAction
+import org.wordpress.android.ui.uploads.UploadActionUseCase.UploadAction.REMOTE_AUTO_SAVE
+import org.wordpress.android.ui.uploads.UploadActionUseCase.UploadAction.UPLOAD
+import org.wordpress.android.ui.uploads.UploadActionUseCase.UploadAction.UPLOAD_AS_DRAFT
 import org.wordpress.android.util.NetworkUtilsWrapper
 import javax.inject.Inject
-import javax.inject.Singleton
 
-@Singleton
+@Reusable
```

Comment:
🎉

Reply:
@maxme Just for the info. I initially annotated the `UploadActionUseCase` with `@Singleton`, because I was injecting it into a Singleton object. I thought it could cause some issues since I'd be injecting a short living object into a long living object.... But as far as I can tell it doesn't cause any issues - the only downside is that the object annotated with `@Reusable` will technically be a Singleton since the garbage collector will never collect it since it will always be referenced from the Singleton object.

Reply:
Thanks for explaining it, I didn't notice where `UploadActionUseCase` was used.

>  the only downside is that the object annotated with @Reusable will technically be a Singleton since the garbage collector will never collect it since it will always be referenced from the Singleton object.

So that's even not a downside compared to `@Singleton`, it's just going to match the behavior where it gets injected. And if this injected singleton is not used but your reusable is used elsewhere, we still get the reusable upsides.  Anyway there is probably no noticeable difference between the 2 for this kind of instances :).

Reply:
💯

Reply:
For `UploadActionUseCase`, I think it would have been okay if it was instantiated normally. The additional memory footprint isn't big. And there is a reduced risk if the class ends up storing state and we forget to remove `@Reusable`.